app/lib/actions/katello/host/update.rb

```diff
@@ -12,11 +12,11 @@ def plan(host, consumer_params = nil)
             host.content_facet.save! if host.content_facet
 
             if host.subscription_facet
-              if consumer_params
-                host.subscription_facet.update_from_consumer_attributes(consumer_params)
-              else
+              unless consumer_params
                 consumer_params = host.subscription_facet.consumer_attributes
               end
+
+              host.subscription_facet.update_from_consumer_attributes(consumer_params)
```

Comment:
Either the consumer_params are passed in, or they are fetched (some attributes are stripped out in controller). This may result in extra updates but it will guarantee everything is in sync.